src/components/Users/Users.js

```diff
@@ -1,15 +1,86 @@
-import User from '../User/User';
-import './Users.css';
-
-const Users = ({ users = [] }) => {
-  return (
-    <article className="Users">
-      {users.map((user) => {
-        const { id } = user;
-        return <User key={id} user={user} />;
-      })}
-    </article>
-  );
+import { useEffect } from "react";
+import User from "../User/User";
+import "./Users.scss";
+
+const Users = ({
+  users = [],
+  searchInput,
+  expanded,
+  setExpanded,
+  loading,
+  error,
+}) => {
+  useEffect(() => {
+    const expandedList = window.localStorage.getItem("usersList");
+    if (expandedList !== null) {
+      setExpanded(JSON.parse(expandedList));
+    }
+  }, []); // eslint-disable-line
```

Comment:
This is very cool! I love that the state will persist from session to session. Nice work here

Reply:
This is just a suggestion you can ignore, but if you are interested in continuing with this it could be fun. The app wants to sync the `expanded` state with localStorage, which means that the developers on your team need to remember to call `setItem` whenever they call `setExpanded`. You've done that very well so far, but eventually some developer on your team will forget to do it. You can make this more convenient for your team by creating a helper function that does both things (call `setExpanded` and write to localStorage). Then developers could use this helper instead of `setExpanded`. Even more interesting, you could create a _custom hook_ that keeps a state variable in sync with local storage.

Reply:
I'll work on that thanks.